test/metabase/query_processor/middleware/expand_macros_test.clj

```diff
@@ -48,6 +48,32 @@
                          [:> [:field 4 nil] 1]]]
              :breakout [[:field 17 nil]]}))))))
 
+(deftest nested-segments-test
+  (t2.with-temp/with-temp
+    [:model/Database {database-id  :id} {}
+     :model/Table    {table-id     :id} {:db_id database-id}
+     :model/Segment  {segment-1-id :id} {:table_id   table-id
+                                         :definition {:filter [:= [:field 5 nil] "abc"]}}
+     :model/Segment  {segment-2-id :id} {:table_id   table-id
+                                         :definition {:filter [:and
+                                                               [:segment segment-1-id]
+                                                               [:is-null [:field 7 nil]]]}}
+     :model/Segment _ {:table_id table-id
+                       :id 10002
```

Comment:
Please don't hardcode IDs like this in `with-temp` forms, this is going to break if we ever write a test that generates a lot of Segments programmatically or if anyone runs it locally on a long-running instance. Capture the auto-generated IDs like you're doing above

Reply:
Whole test was refactored and IDs are obtained the correct way now.